cpp/include/raft/linalg/transpose.cuh

```diff
@@ -55,7 +56,95 @@ void transpose(math_t* inout, int n, cudaStream_t stream)
   detail::transpose(inout, n, stream);
 }
 
+/**
+ * @brief Transpose a contiguous matrix. The output has same layout policy as the input.
```

Comment:
It doesn't have to be contiguous since we can handle submatrix. Maybe we can relax the constraint by allowing certain types of strides? Suggestions are welcomed.